packages/ebayui-core/src/components/ebay-video/component.ts

```diff
@@ -392,13 +475,68 @@ class Video extends Marko.Component<Input, State> {
             );
         });
 
+        if(this.input.offscreenPause) {
+            // Set up Intersection Observer to detect when video is 50% in viewport
+            this.setupIntersectionObserver();
+        }
+
+       
+
         this._loadVideo();
     }
 
+    setupIntersectionObserver() {
+        // Create options for the observer
+        const options = {
+            root: null, // Use the viewport as the root
+            rootMargin: '0px',
+            threshold: 0.5 // 50% visibility threshold
+        };
+
+        // Create the observer
+        this.observer = new IntersectionObserver((entries) => {
+            entries.forEach(entry => {
+                if(this.userPaused) {
+                    // If user has manually paused, do not auto-play/pause
+                    return;
+                }
+                
+                // Auto-play when 50% visible and pause when less than 50% visible
+                if (entry.isIntersecting) {
+                    // Only auto-play if the video is loaded and not already playing
+                    if (this.state.isLoaded && !this.state.failed && this.video.paused) {
+                        // Set isAutoPlay flag to true for tracking
+                        this.isAutoPlay = true;
```

Comment:
I do not understand why we need a `this` variable for autoplay. 
If we want to track if a video is played or paused automatically, we should pass that argument into the `play` or `pause` functions directly from here. 
That said, as I mentioned before, we can also emit an event for that which is probably easier to track.